### DISCUSSION TITLE
how to create container on micro-sd?

### DISCUSSION THREAD

**Discussion by whyakari** [RuriOSS/daijin#27] [Q&A] [answered]
Good evening, good morning, good afternoon, sir. how are u?

well, I have a question, can I mount/create this script on the micro-sd? I have root, and I have a modified kernel and custom rom (I already used docker lol)

**Comment by Moe-hacker** [comment, marked as answer]
> Good evening, good morning, good afternoon, sir. how are u?
> 
> well, I have a question, can I mount/create this script on the micro-sd? I have root, and I have a modified kernel and custom rom (I already used docker lol)

The script can't use micro-sd directly, you have to do some changes of it by yourself. Here's the way to do that(using termux-container v8):
## Preparation:
If your phone has a micro-sd installed, there will be a device called mmcblkxx, on my phone, it's /dev/block/mmcblk0, and /dev/block/mmcblk0p1 is the first partition of it.
If the partition isn't using ext4 filesystem, it's recommended to format it to ext4 by mkfs first.
## Then, mount the micro-sd to your system:
```
mkdir test
mount /dev/block/mmcblk0p1 ./test
```
If the command doesn't work,you can try to mount it as loop device:
```
LOOP=$(losetup -f)
losetup $LOOP /dev/block/mmcblk0p1
mount $LOOP ./test
```
## Edit the script:
If the container path already exists, termux-container will cancel the installation of container, but you can't mount micro-sd without an existing mountpoint,so you need to edit the script:
Find the following lines, and delete them:
```
if [[ -e $CONTAINER_DIR ]];then
    whiptail --title "[] ERROR" --msgbox "Directory already exists!" 7 $WIDTH
    exit 1
  fi
```
Then you can run termux-container and use the path "test" to create the container.
## Edit the script to mount micro-sd automatically before running the container:
Find the following line:
```
RUN_CHROOT_CONTAINER(){
```
and add these lines after it:
```
mount /dev/block/mmcblk0p1 /path/to/test
if [[ $? != 0 ]];then
LOOP=$(losetup -f)
losetup $LOOP /dev/block/mmcblk0p1
mount $LOOP ./test
fi
```

**Reply by whyakari** [reply]
@Moe-hacker, thank you :v
I'll test it and I'll tell you. Thank you very much 🥹

**Reply by whyakari** [reply]
very good! it worked right.
